lib/api/core/hotelClerk.js

```diff
@@ -134,13 +136,13 @@ class HotelClerk {
   countSubscription (request) {
     const
       roomId = request.input.body.roomId,
-      room = this.rooms[roomId];
+      room = this.rooms.get(roomId);
 
-    if (room === undefined) {
+    if (room === undefined || room === null) {
       realtimeError.throw('room_not_found', roomId);
     }
 
-    return {count: this.rooms[roomId].customers.size};
+    return {count: room.customers.size};
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "room" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2259)